### PR TITLE
Rework Marathon UI fixes

### PR DESCRIPTION
**Pull request by clems4ever**
I split and fixed the features that have been developed in the Criteo version of the Marathon UI.

This PR introduces two new columns in the main page: "Logs" and "Service" which are, for each application, filled with a link to the application logs (customizable in the runtime configuration) and a link to the service endpoint.

### CODE REVIEW

**Comment by clems4ever** [comment]
image

You can see the two new columns here. I also fixed big bugs in the UI compared to our current version.

File: src/js/components/AppListItemComponent.jsx [hunk 1]
```diff
@@ -340,6 +360,26 @@ var AppListItemComponent = React.createClass({
     );
   },
 
+  getServiceLink: function () {
+    var model = this.props.model;
+    var scheme = (model.labels && "SERVICE_SCHEME" in model.labels &&
+       (model.labels["SERVICE_SCHEME"] === "http" ||
+        model.labels["SERVICE_SCHEME"] === "https"))
```
Comment (kamaradclimber) [inline src/js/components/AppListItemComponent.jsx:367]: at the moment we dont have any service exposed on https (but soon!)

File: src/js/components/AppListItemComponent.jsx [hunk 2]
```diff
@@ -319,6 +325,20 @@ var AppListItemComponent = React.createClass({
     return <i className="icon icon-small app" title="Application"></i>;
   },
 
+  getLogsLink: function () {
+    var model = this.props.model;
+    const logsLink = Mustache.render(Config.appLogsLinkPattern, {
```
Comment (kamaradclimber) [inline src/js/components/AppListItemComponent.jsx]: I'm not sure to understand why you define this link outside of the condition

File: src/js/config/runtimeConfig.template.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+var runtimeConfig = {
+  // The domain used to compute the service endpoint
+  serviceDomain: "marathon-par.central.criteo.preprod",
+
+  // The pattern to build the links to the logs
+  appLogsLinkPattern: "http://kibana.par.preprod.crto.in/app/kibana#/discover?\
```
Comment (kamaradclimber) [inline src/js/config/runtimeConfig.template.js]: criteo specific urls. Do you need to have these in open source code?

**Review by kamaradclimber — CHANGES_REQUESTED** [review]
Super useful feature. I think it would be better to avoid committing criteo urls on github if we can avoid it)

File: src/js/config/runtimeConfig.template.js (hunk 3 above)
Comment (pierresouchay) [reply src/js/config/runtimeConfig.template.js]: And furthermore those URLs do work only for PAR and prod

File: src/js/components/AppListItemComponent.jsx (hunk 2 above)
Comment (clems4ever) [reply src/js/components/AppListItemComponent.jsx]: Ack.

File: src/js/components/AppListItemComponent.jsx (hunk 1 above)
Comment (clems4ever) [reply src/js/components/AppListItemComponent.jsx:367]: We won't have to wonder then :).

File: src/js/config/runtimeConfig.template.js (hunk 3 above)
Comment (clems4ever) [reply src/js/config/runtimeConfig.template.js]: @kamaradclimber , you are right, I will replace it with something unrelated like http://mylogs/?appId={{appId}}. This is just a template.

@pierresouchay, this "pattern" (that I will rename "template" btw) is supposed to be different in each DC and env. This is why I leveraged the concept of runtimeConfig brought by Cedric. That way, we won't have to rebuild the package for every (dc, env).

File: src/js/components/AppListItemComponent.jsx [hunk 4]
```diff
@@ -340,6 +360,26 @@ var AppListItemComponent = React.createClass({
...
   },
 
+  getServiceLink: function () {
+    var model = this.props.model;
+    var scheme = (model.labels && "SERVICE_SCHEME" in model.labels &&
+       (model.labels["SERVICE_SCHEME"] === "http" ||
+        model.labels["SERVICE_SCHEME"] === "https"))
+      ? model.labels["SERVICE_SCHEME"]
+      : "http";
+    var cleanAppName = model.id.substring(1).replace("/", "-");
```
Comment (clems4ever) [inline src/js/components/AppListItemComponent.jsx:370]: The clean app name is not very generic but I don't see a cleaner way to compute the service link yet.

**Comment by clems4ever** [comment]
Everything should be fixed now.

**Comment by clems4ever** [comment]
BTW, I wanted to add a link to the logs per task but we don't have the task id in kibana yet.

File: src/js/components/AppListItemComponent.jsx (hunk 4 above)
Comment (kamaradclimber) [reply src/js/components/AppListItemComponent.jsx:370]: that's seem generic enough for the moment

**Review by kamaradclimber — APPROVED** [review]

**Pull request merged**